packages/hardhat-core/src/internal/hardhat-network/provider/transactions/FakeSenderTransaction.ts

```diff
@@ -100,18 +98,14 @@ export class FakeSenderTransaction extends Transaction {
   private readonly _sender: Address;
 
   constructor(sender: Address, data: TxData = {}, opts?: TxOptions) {
-    const fakeSignature: Buffer = crypto
-      .createHash("md5")
-      .update(Buffer.from(`${util.inspect(data)}`))
-      .update(sender.buf)
-      .digest();
+    const fakeSignature = makeFakeSignature(data, sender);
 
     super(
       {
         ...data,
-        v: data.v ?? 27,
-        r: data.r ?? fakeSignature.readUInt32LE(),
-        s: data.s ?? fakeSignature.readUInt32LE(4),
+        v: data.v ?? fakeSignature.v,
```

Comment:
Just a reminder that we should use the current value here.

Reply:
Addressed in https://github.com/NomicFoundation/hardhat/pull/3175/commits/524af2d7e770ad3663d1660a55b309ac175cb253